src/main/webapp/js/datahandlers/singlepagedatahandler.js

```diff
@@ -45,16 +44,28 @@ class SinglePageDataHandler {
   }
 
   /**
+   * @param {Object} jsonResponse
+   * @returns {Object} Json data needed for rendering page.
+   * @private
+   */
+  getJsonData_(jsonResponse) {
+    return jsonResponse;
+  }
+
+  /**
+   * @param {string} id 
    * @returns {string} path
+   * @private
    * @abstract
    */
-  getRequestPath_() {}
+  getRequestPath_(id) {}
 
     /**
    * This method converts from scholarship JSON object to a JS object map, 
    *  which will be used to render the scholarship page soy template.
    * @param {*} data - The JSON object to be converted.
    * @return {Object} - The object map representing a scholarship's data.
+   * @private
```

Comment:
Flip @abstract and @private. The order usually goes params, return, everything else, visibility.